src/helpers/api.js

```diff
@@ -0,0 +1,36 @@
+const ATTRIBUTES_ENDPOINT = 'attributes.json'
+
+let d2
+let api
+
+/**
+ * Sets d2 and the api
+ */
+export const initApi = d2Instance => {
+    d2 = d2Instance
+    api = d2.Api.getApi()
+}
+
+export const getAttributes = ({
+    filter = [],
+    paging = false,
+    unique = true,
+}) => {
+    const params = [
+        `paging=${paging}`,
+        `fields=${['id', 'displayName'].join(',')}`,
+        ...[...filter, `unique:eq:${unique}`].map(f => `filter=${f}`),
```

Comment:
I think that this is a little difficult to scan. Maybe moving a part of it out into the function body would help